docs/guides/plugin_dev/workflows.rst

```diff
@@ -0,0 +1,131 @@
+
+.. _kernel_config:
+
+Preparing a model workflow
+==========================
+
+A user can define a complete pipeline of methods and their options to :ref:`build a model <model_build>` or :ref:`updating an existing model <model_update>`
+in the configuration `.yaml file <https://en.wikipedia.org/wiki/YAML>`_
+
+**Steps in brief:**
+
+1) Start with a template of the :ref:`HydroMT model plugin <plugins>` with which you want to build / update a model. Templates can be found in the online documentation of each model plugin.
+2) Edit / add / remove sections (i.e. methods) based on which components you want to build or adapt. The arguments of specific methods can be found in the API chapter in the online documentation of each model plugin.
+3) Save the configuration file and use it in combination with the HydroMT :ref:`build <model_build>` and :ref:`update <model_update>` methods.
+
+.. NOTE::
+
+    The HydroMT configuration file used to be in ini format, this has been deprecated and can no longer be used.
+    The new format (supported from version 0.7.1) is a yaml file, which is more flexible and easier to read and write.
+
+.. NOTE::
+
+    The HydroMT model configuration (.yaml) file should not be confused with the simulation configuration file of the model kernel.
+    While the first defines how HydroMT should build or update a model, the latter defines the simulation for the model kernel.
+    The format of the latter differs with each plugin, but can be accessed in HydroMT trough the :py:meth:`~hydromt.Model.config` component.
+
+
+Workflow (.yaml) file
+--------------------------------
+
+The YAML file serves as the configuration and workflow definition for building and updating models in HydroMT. It uses a simple key-value syntax with structured sections to define steps and arguments for each method. Each step corresponds to a model action (method), and arguments specify how that action is applied. The order of execution is determined by the sequence of steps listed in the YAML file.
+
+- **Model Type**: The top level defines the `modeltype`, such as "model".
+- **Global Components**: Under the `global` section, you can define reusable components
+  that apply throughout the model workflow. Each component is named and has a specific
+  type, like `GridComponent` or `ConfigComponent`.
+- **Steps**: The main logic of the model is defined under `steps`. Each step is a method call, which typically includes the method name and its required arguments. The order of the steps is critical, as HydroMT executes each step sequentially.
+
+Below is an example of the YAML format used in HydroMT:
+
+```yaml
```

Comment:
maybe test parsing this using doctest?

Reply:
We can try to parse this, but I'm hesitant to actually execute it. Because of all they dynamic imports/inspection etc, there is no way to meaningfully execute it aside from actually doing the model building, which would very significantly slow down the docs generation. 

Over all I'm very on the fence, becasue having this tested constantly would be very good. Perhaps we can make it a bit shorter to try and make sure the exec times stay low enough. What do you think?

Reply:
I just want to make sure that the posted yaml is valid.
If we do not have a validation step outside of executing, then I would skip it.